catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -320,26 +334,100 @@ private String getBlackListAsRegex() {
+        Thread.currentThread().interrupt();
+        return;
+      } catch (IllegalStateException e) {
+        LOGGER.info(
+            "CDM config for {} deleted while waiting for InputTransformers, CDM camel route won't be started",
+            monitoredDirectory);
+        return;
+      }
+
+      LOGGER.trace(
+          "Attempting to add routes for content directory monitor watching {}", monitoredDirectory);
+      try {
+        routeBuilder = createRouteBuilder();
+        verifyCamelComponentIsAvailable("content");
+        verifyCamelComponentIsAvailable("catalog");
+        camelContext.addRoutes(routeBuilder);
+      } catch (Exception e) {
+        LOGGER.info(
+            "Unable to configure Camel route. Resources will not be ingested. CDM configuration may still exist and should be remade",
+            e);
+      } finally {
+        if (LOGGER.isDebugEnabled()) {
+          dumpCamelContext("after attemptAddRoutes()");
+        }
+      }
+    }
+  }
+
+  /**
+   * Should be called when synchronized under the {@link #lock} object.
+   *
+   * @throws IllegalStateException if the CDM configuration was deleted while waiting for
+   *     InputTransformers
+   * @throws InterruptedException if the thread was interrupted while waiting for InputTransformers
+   */
+  @SuppressWarnings({
+    "squid:S2273" /* Method is called from within a synchronized block */,
+    "squid:S2209" /* Accessing InputTransformerIds static reference for testing */
+  })
+  private void waitForInputTransformers() throws InterruptedException {
+    final String propertyId = "id";
+    Set<String> propertyValues = new HashSet<>(inputTransformerIds.getTransformerIds());
+
+    if (propertyValues.isEmpty()) {
+      return;
+    }
+
+    long lastChecked = new Date().getTime();
+    long end = lastChecked + transformerWaitTimeoutMillis;
+    for (; ; ) {
+      if (configDeleted.get()) {
+        throw new IllegalStateException();
+      }
+
+      for (ServiceReference serviceReference : inputTransformers) {
+        Object propertyValue = serviceReference.getProperty(propertyId);
+        if (propertyValue instanceof String) {
+          propertyValues.remove(propertyValue);
+          LOGGER.debug("Remaining property values {}", propertyValues);
+        }
+      }
+
+      if (propertyValues.isEmpty()) {
+        LOGGER.trace("Finished finding all InputTransformers");
+        break;
+      }
+
+      if (lastChecked >= end) {
+        LOGGER.warn(
+            "Not all input transformers were found, starting up CDM route anyways. Ingested resources may not have expected metadata");
+        break;
+      } else {
+        LOGGER.trace(
+            "Waiting {} seconds before checking InputTransformers again",
+            TimeUnit.MILLISECONDS.toSeconds(transformerCheckPeriodMillis));
+        lastChecked += transformerCheckPeriodMillis;
+        lock.wait(transformerCheckPeriodMillis);
```

Comment:
Even if the caller of the method is acquiring the lock, you should also acquire it in here to make it clear you are. Doing a `wait()` on an object in a method without first synchronizing on it is not a good practice and can easily lead to issues during maintenance if a developer decides to remove the synchronization in the calling method.